code/game/gamemodes/cult/cult_comms.dm

```diff
@@ -62,16 +62,111 @@
 	popup.set_content(text)
 	popup.open()
 	return 1
+	
+/mob/living/proc/cult_master()
+	set category = "Cultist"
+	set name = "Assert Leadership"
+	pollCultists()
+	return
+
+/datum/action/innate/cultmast
+	name = "FinalReckoning"
+	button_icon_state = "sintouch"
+	background_icon_state = "bg_demon"
+	buttontooltipstyle = "cult"
+	check_flags = AB_CHECK_RESTRAINED|AB_CHECK_STUNNED|AB_CHECK_CONSCIOUS
+
+/datum/action/innate/cultmast/IsAvailable()
+	if(owner.mind.special_role != "Cult Master")
+		return 0
+	return ..()
+
+/datum/action/innate/cultmast/Activate()
+	for(var/i in 1 to 4)
+		if(do_after(owner, 30, target = owner))
+			for(var/mob/M in mob_list)
+				if(iscultist(M) && M != owner)
+					var/turf/mobloc = get_turf(M)
+					switch(i)
+						if (1)
+							new /obj/effect/overlay/temp/cult/sparks(mobloc, M.dir)
+							playsound(mobloc, "sparks", 50, 1)
+						if (2)
+							new /obj/effect/overlay/temp/dir_setting/cult/phase/out(mobloc, M.dir)
+							playsound(mobloc, "sparks", 75, 1)
+						if (3)
+							new /obj/effect/overlay/temp/dir_setting/cult/phase(mobloc, M.dir)
+							playsound(mobloc, "sparks", 100, 1)
+						if (4)
+							new /obj/effect/overlay/temp/cult/spray (mobloc)
+							playsound(mobloc, 'sound/magic/lightningbolt.ogg', 100, 1)
+							mobloc = get_turf(owner)
+							M.forceMove(mobloc)
+		if(i == 4)
+			for(var/datum/action/innate/cultmast/H in owner.actions)
+				qdel(H)
+
+
+/datum/action/innate/cultmark
+    name = "Mark"
+    desc = "Marks a target for the cult"
+    check_flags = AB_CHECK_RESTRAINED|AB_CHECK_STUNNED|AB_CHECK_CONSCIOUS
+    var/obj/effect/proc_holder/cultmark/CM
+
+/datum/action/innate/cultmark/New()
+    CM = new()
+    ..()
+
+/datum/action/innate/cultmark/Destroy()
+    qdel(CM)
+    CM = null
+    return ..()
+
+/datum/action/innate/cultmark/Activate()
+    CM.toggle(owner) //the important bit
+    return TRUE
+
+/obj/effect/proc_holder/cultmark
+    active = FALSE
+    ranged_mousepointer = 'icons/effects/cult_target.dmi'
+
+/obj/effect/proc_holder/cultmark/proc/toggle(mob/user) //the important bit
+    if(active)
+        remove_ranged_ability("You cease the marking ritual...")
+    else
+        add_ranged_ability(user, "You prepare to mark a target for your cult...")
+
+/obj/effect/proc_holder/cultmark/InterceptClickOn(mob/living/caller, params, atom/target)
+	if(..())
+		return
+	if(ranged_ability_user.incapacitated())
+		remove_ranged_ability()
+		return
+	var/turf/T = ranged_ability_user.loc
+	if(!isturf(T))
+		return FALSE
+	if(target in view(7, get_turf(ranged_ability_user)))
+		blood_target = target
+		for(var/mob/M in mob_list)
+			if(iscultist(M))
+				M << "<span class='cultlarge'><b>["Master"] [caller] has marked [blood_target] as the cult's highest priority, get there immediately!</b></span>"
+		remove_ranged_ability(caller, "The marking rite is complete! Your mark will last for 2 minutes or until a new mark is added")
+		spawn(120)
+			blood_target = null
```

Comment:
Need some sanity checks here, to make sure the mark you're removing is the target that was marked 2 minutes ago; this should probably be a global proc with a unique addtimer instead of a spawn, eg;

`addtimer(CALLBACK(GLOBAL_PROC, .proc/reset_blood_target, blood_target), 120, TIMER_OVERRIDE)`

    /proc/reset_blood_target(old_target)
    	if(blood_target != old_target)
    		return
     	for(var/mob/M in mob_list)
    		if(iscultist(M))
    			M << "<span class='cultlarge'><b>The blood mark on [blood_target] has expired!</b></span>"
     	blood_target = null